### PR TITLE
Add license headers on build bundles

### LINKED ISSUE #10458: License headers are missing on built bundles
Our built files are missing license headers. 

cjs has no headers:
https://unpkg.com/react@16.0.0-beta.5/cjs/react.development.js
https://unpkg.com/react-dom@16.0.0-beta.5/cjs/react-dom.development.js

umd has headers identifying the bundle name but not the license:
https://unpkg.com/react@16.0.0-beta.5/umd/react.development.js
https://unpkg.com/react-dom@16.0.0-beta.5/umd/react-dom.development.js

Both should have license headers similar to this:

```
/**
 * React v15.6.1
 *
 * Copyright 2013-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 */
```

We likely just need to change the code in getBanner in scripts/rollup/build.js. https://github.com/facebook/react/blob/b4a3e3b34594759c90aef1694d2166b2d1126fa1/scripts/rollup/build.js#L78

You can test by running `yarn build` and looking in build/packages.

cc @trueadm

### PR DESCRIPTION
**Pull request by gzhappysky**
fix #10458

### CODE REVIEW

File: scripts/rollup/header.js [hunk 1]
```diff
@@ -15,14 +15,22 @@ function getProvidesHeader(hasteFinalName) {
 `;
 }
 
-function getUMDHeader(filename, reactVersion) {
+function getHeader(filename, reactVersion) {
   return `/**
- * ${filename} v${reactVersion}
+ * React v${reactVersion}
```
Comment (gaearon) [inline scripts/rollup/header.js]: We should still include the filename IMO. Maybe make it the first line, and then the rest?

Comment (gzhappysky) [inline scripts/rollup/header.js]: I feel version more important! Might we can do it like this?  

 /**  
 * React v16.0.0-beta.5
 * react.development.js
 *
 * Copyright 2013-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 */

**Comment by gzhappysky** [comment]
done. Please review. @gaearon

File: scripts/rollup/header.js [hunk 2]
```diff
@@ -15,14 +15,23 @@ function getProvidesHeader(hasteFinalName) {
 `;
 }
 
-function getUMDHeader(filename, reactVersion) {
+function getHeader(filename, reactVersion) {
   return `/**
```
Comment (Daniel15) [inline scripts/rollup/header.js]: I wonder if this should have an exclamation mark (like `/**!`) as that signals to most minifiers that it should not be removed when minifying the code.

Comment (gzhappysky) [inline scripts/rollup/header.js]: cc @gaearon

Comment (gaearon) [reply scripts/rollup/header.js]: I think it's fine to leave as is, we didn't do this before either.

File: scripts/rollup/build.js [hunk 3]
```diff
@@ -80,12 +80,14 @@ function getBanner(bundleType, hasteName, filename) {
...
     // CommonJS DEV bundle is guarded to help weak dead code elimination.
     case NODE_DEV:
+      let banner = Header.getHeader(filename, reactVersion);
       // Wrap the contents of the if-DEV check with an IIFE.
       // Block-level function definitions can cause problems for strict mode.
-      return `'use strict';\n\n\nif (process.env.NODE_ENV !== "production") {\n(function() {\n`;
+      banner += `'use strict';\n\n\nif (process.env.NODE_ENV !== "production") {\n(function() {\n`;
+      return banner;
     case NODE_PROD:
       return '';
```
Comment (gaearon) [inline scripts/rollup/build.js]: Let's not forget `NODE_PROD` too.

Comment (gzhappysky) [reply scripts/rollup/build.js]: I have added license header to NODE_PROD env, Seems to be filtered by [this](https://github.com/facebook/react/blob/master/scripts/rollup/build.js#L345).

Comment (gaearon) [reply scripts/rollup/build.js]: Right. So maybe there's some way to tell it to preserve the comment with some special syntax or `@license` word in it.

**Pull request merged**